src/GraphQL.Server.AspNetCore.GraphiQL/Internal/GraphiQLPageModel.cs

```diff
@@ -0,0 +1,34 @@
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace GraphQL.Server.AspNetCore.GraphiQL.Internal {
+
+	// https://docs.microsoft.com/en-us/aspnet/core/mvc/razor-pages/?tabs=netcore-cli
+	internal class GraphiQLPageModel {
+
+		private static string graphiQLCSHtml;
+
+		private readonly GraphiQLMiddlewareSettings settings;
+
+		public GraphiQLPageModel(GraphiQLMiddlewareSettings settings) {
+			this.settings = settings;
+		}
+
+		public string Render() {
+			if (graphiQLCSHtml != null) {
+				return graphiQLCSHtml;
+			}
+			var assembly = typeof(GraphiQLPageModel).GetTypeInfo().Assembly;
+			var resource = assembly.GetManifestResourceStream("GraphQL.Server.AspNetCore.GraphiQL.Internal.graphiql.cshtml");
```

Comment:
Does this stream need disposing?

Reply:
Yes, disposing the stream and the reader